### PR TITLE
fpl tests

### PR DESCRIPTION
**Pull request by clewis7**
refactored so there are no large binary files

did reorg for names

#152

### CODE REVIEW

File: examples/gridplot/gridplot_cmap.py
Comment (kushalkolar) [inline examples/gridplot/gridplot_cmap.py:1]: I guess we can just keep it to make sure accessing graphics within a gridplot works etc.

File: examples/imagewidget/iw_simple.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    data=data,
+    slider_dims=["t"],
+    vmin_vmax_sliders=True,
+    cmap="gnuplot2"
+)
+
+data = np.load("../data/random.npy")
+
+iw.show()
+
```
Comment (kushalkolar) [inline examples/imagewidget/iw_simple.py:33]: can you test before and after setting timepoint?

set timepoint using the slider  as well as `@current_index`

File: examples/imagewidget/iw_simple.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+iw["t"] = 250
```
Comment (kushalkolar) [inline examples/imagewidget/iw_simple.py:34]: I don't think this is possible.

File: examples/imagewidget/iw_simple.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+iw.plot.center_scene()
+
```
Comment (kushalkolar) [inline examples/imagewidget/iw_simple.py:37]: also check that window functions and frame_apply works

File: examples/scatter/scatter.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+# create a random distribution of 10,000 xyz coordinates
+n_points = 10_000
+
+# if you have a good GPU go for 1.5 million points :D
+# this is multiplied by 3
+n_points = 500_000
+
+# dimensions always have to be [n_points, xyz]
+dims = (n_points, 3)
```
Comment (kushalkolar) [inline examples/scatter/scatter.py:29]: you don't need all these since you're loading existing data

File: examples/line/present.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+from pygfx import WgpuRenderer
+
+canvas = WgpuCanvas()
+renderer = WgpuRenderer(canvas)
+
+plot = Plot(canvas=canvas, renderer=renderer)
+
+sine = np.load("../data/sine.npy")
+cosine = np.load("../data/cosine.npy")
+sinc = np.load("../data/sinc.npy")
```
Comment (kushalkolar) [inline examples/line/present.py:23]: just make these, don't save them since it's not randomly generated

File: examples/line/line.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+from pygfx import WgpuRenderer
+
+canvas = WgpuCanvas()
+renderer = WgpuRenderer(canvas)
+
+plot = Plot(canvas=canvas, renderer=renderer)
+
+sine = np.load("../data/sine.npy")
+cosine = np.load("../data/cosine.npy")
+sinc = np.load("../data/sinc.npy")
```
Comment (kushalkolar) [inline examples/line/line.py:23]: just make these, don't save them since it's not randomly generated

File: examples/line/dataslice.py [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+from pygfx import WgpuRenderer
+
+canvas = WgpuCanvas()
+renderer = WgpuRenderer(canvas)
+
+plot = Plot(canvas=canvas, renderer=renderer)
+
+sine = np.load("../data/sine.npy")
+cosine = np.load("../data/cosine.npy")
+sinc = np.load("../data/sinc.npy")
```
Comment (kushalkolar) [inline examples/line/dataslice.py:23]: just make these, don't save them since it's not randomly generated

File: examples/line/colorslice.py [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+from pygfx import WgpuRenderer
+
+canvas = WgpuCanvas()
+renderer = WgpuRenderer(canvas)
+
+plot = Plot(canvas=canvas, renderer=renderer)
+
+sine = np.load("../data/sine.npy")
+cosine = np.load("../data/cosine.npy")
+sinc = np.load("../data/sinc.npy")
```
Comment (kushalkolar) [inline examples/line/colorslice.py:23]: just make these, don't save them since it's not randomly generated

File: examples/line/present.py
Comment (kushalkolar) [inline examples/line/present.py:1]: this tests both present and scaling

Comment (kushalkolar) [reply examples/line/present.py:1]: so you can just add that to the comment and filename

**Review by kushalkolar — CHANGES_REQUESTED** [review]

**Comment by kushalkolar** [comment]
it's not working just through pytest because:

1. you need an `__init__.py` in `examples/tests`
2. You need relative import for testutils, `from .testutils import` instead of `from testutils import`
3. you need relative paths to data files, for example:
```python
data_path = Path(__file__).parent.parent.joinpath("data", "random.npy")
data = np.load(data_path)
```

**Comment by kushalkolar** [comment]
Seems like zenodo is still the best place to store these, ah well. The file will be small at least so download time should be very fast.

File: examples/line/present.py (hunk 5 above)
Comment (clewis7) [reply examples/line/present.py:23]: when I didn't save them was when I kept getting errors locally about the data changing...idk what the precision is like when you call `np.sin()` or `np.cos()` on a linspace but I think this is what was causing the problem before and that is why I saved them

Comment (kushalkolar) [reply examples/line/present.py:23]: what if you force `float32` when making the linspace?

Comment (clewis7) [reply examples/line/present.py:23]: will try

File: examples/imagewidget/iw_simple.py (hunk 1 above)
Comment (clewis7) [reply examples/imagewidget/iw_simple.py:33]: need to think some more about testing image widget, somehow need renderer and canvas to be offscreen as in the other examples

**Comment by kushalkolar** [comment]
superseeded by #179

**Pull request closed without merging**